application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/AttachmentListener.java

```diff
@@ -0,0 +1,126 @@
+package org.togetherjava.tjbot.commands.moderation.attachment;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.AuditableRestAction;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.tools.StringUtils;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent messages in every channel and checks if the message has any
+ * blacklisted file attached.
+ * <p>
+ * If there was a blacklisted file attached, delete the message and send the user a dm, telling what
+ * they did wrong and also inform the mods.
+ */
+public class AttachmentListener extends MessageReceiverAdapter {
+
+    private final Config config;
+
+    private final ModAuditLogWriter modAuditLogWriter;
+
+    /**
+     * Creates a AttachmentListener to receive all message sent in any channel.
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public AttachmentListener(@NotNull Config config,
+            @NotNull ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.config = config;
+        this.modAuditLogWriter = modAuditLogWriter;
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (attachmentContainsBlacklistedFileExtension(event)) {
+            deleteMessage(event).flatMap(any -> dmUser(event)).queue();
+            warnMods(event);
+        }
+    }
+
+    private AuditableRestAction<Void> deleteMessage(@NotNull MessageReceivedEvent event) {
+        return event.getMessage().delete();
+    }
+
+    private RestAction<Message> dmUser(@NotNull MessageReceivedEvent event) {
+        return dmUser(event.getMessage(), event.getAuthor().getIdLong(), event.getJDA());
+    }
+
+    private RestAction<Message> dmUser(Message originalMessage, long userId, @NotNull JDA jda) {
+        Message dmMessage = createDmMessage(originalMessage);
+        return jda.openPrivateChannelById(userId)
+            .flatMap(channel -> channel.sendMessage(dmMessage));
+    }
```

Comment:
call the second method directly, can't see a need for wrapper method

Reply:
in my last PR @Zabuzard requested this behavior --> https://github.com/Together-Java/TJ-Bot/pull/518#discussion_r949982321